metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -247,6 +257,21 @@ class MetalsLanguageServer(
       InitializationOptions.Default
     )
 
+  def parseTreesAndPublishDiags(paths: Seq[AbsolutePath]): Future[Seq[Unit]] = {
+    Future.sequence {
+      paths.distinct.map { path =>
+        if (path.isScalaFilename) {
+          // diagnostics from the compiler are still needed for Scala 3
+          compilers.didChange(path).map { diags =>
+            diagnostics.onSyntaxError(path, diags ++ trees.didChange(path))
+          }
+        } else {
+          Future.successful(())
+        }
+      }
+    }
+  }
```

Comment:
not enough traverse 😁

```suggestion
    Future.traverse(paths.distinct) { path =>
      if (path.isScalaFilename) {
        // diagnostics from the compiler are still needed for Scala 3
        compilers.didChange(path).map { diags =>
          diagnostics.onSyntaxError(path, diags ++ trees.didChange(path))
        }
      } else {
        Future.successful(())
      }
    }
  }
```

Reply:
It was `traverse` all along!

Reply:
Fixed!